### ISSUE TITLE
SV Calling for population 

### ISSUE THREAD

**Issue by MehmetGoktay** [fritzsedlazeck/Sniffles#155]
Hi Fritz, 

I have followed your pipeline which was documented here. https://github.com/fritzsedlazeck/Sniffles/wiki/SV-calling-for-a-population.

I have got final merged vcf file which was genotyped, Like below

./SURVIVOR merge vcf_files_gt_calls.txt 1000 -1 1 -1 -1 -1 merged_gt_SURVIVOR_1kbpdist_typesave.vcf

My question is : why I dont get a single genotype as  0/0 for any svs? I only get ./. and 1/1 genotypes. 

Is it normal or should I interpret missing data ./. as matches to the reference like 0/0?

Best,
Mehmet

**Comment by fritzsedlazeck** [comment]
Mhm if you see ./. this means he did not find a single ref or alt supporting read. 
So please make sure that the chromosome names are matching between your input files. 
And let me know if that helps.
Thanks
Fritz

**Comment by MehmetGoktay** [comment]
I see. So it is supposed to give 0/0 if enough data supports it.  Then there is something wrong then because I did not get even a single 0/0. 

I will do the analysis again may be something went wrong during the analysis. 

Best,
Mehmet

**Comment by fritzsedlazeck** [comment]
Yes, please keep me posted. This part of the code is a newer part so it can be that there is something going on also on my side. 
Thanks
Fritz

**Comment by fritzsedlazeck** [comment]
Oh wait a sec. 
So you are saying you get ./. after SURVIVOR? That is expected. You need to rerun Sniffles with --Ivcf to obtain the genotypes for the individual samples and then merge again. 
Sorry I misread this. Just woke up... 
Thanks
Fritz

**Comment by MehmetGoktay** [comment]
Actually I did it. 

I followed this steps:

1. Obtain calls for each of the samples

This is done by running Sniffles on every sample. For example:

./sniffles -m my_sample.bam -v my_sample.vcf
2. Merge all the vcf files across all samples

First, obtain SURVIVOR from here: SURVIVOR_github

Please sort the files prior to merging.

Second, we need to put all file paths over all vcf files together e.g.:

ls *sample.vcf > vcf_files_raw_calls.txt

Third, we call SURVIVOR to merge these into one vcf file:

./SURVIVOR merge vcf_files_raw_calls.txt 1000 1 1 -1 -1 -1 merged_SURVIVOR_1kbpdist_typesave.vcf

This will generate one VCF file for all the samples, but we lack the information if a SV identified in one sample but not in the other is really absent.
3. Force call all the SVs across all the samples

Next we ran Sniffles again across all the samples similar to this:

./sniffles -m my_sample.bam -v my_sample_gt.vcf --Ivcf merged_SURVIVOR_1kbpdist_typesave.vcf

After this is run over each sample we merge them again to obtain now a fully genotyped multisample vcf:

ls *sample_gt.vcf > vcf_files_gt_calls.txt

and

./SURVIVOR merge vcf_files_gt_calls.txt 1000 -1 1 -1 -1 -1 merged_gt_SURVIVOR_1kbpdist_typesave.vcf

So this final file "merged_gt_SURVIVOR_1kbpdist_typesave.vcf" has only ./. and 1/1.

**Comment by fritzsedlazeck** [comment]
Ok that is good. Sorry I just wanted to make sure that you didnt stop at step 2. 
I should make a python script or something that this is easier, but I did not anticipate that many people will require it. 
Please let me know if you find anything or if that problem seems to be on my side.
Thanks

**Comment by MehmetGoktay** [comment]
Hi Fritz,

Unfortunately still problem remains. Still I only get 1/1 and ./. genotypes. It does not produce any 0/0 genotypes. 

I have followed exactly the same pipeline as you explained for population calls.

1) I called individual svs with sniffless
2) I sorted all vcf files (vcf-sort)
3) merged them with survivor
4) Then forced them to call genotypes against merged set (./sniffles -m my_sample.bam -v my_sample_gt.vcf --Ivcf merged_SURVIVOR_1kbpdist_typesave.vcf)
After this step I never get any matches to reference for any SVs. Only ./. and 1/1. 

I think this is on your side.

Best,
Mehmet

**Comment by fritzsedlazeck** [comment]
Yes sorry I haven't pushed the code yet. Wanted to run some more tests today.
I will ping you once I did.
Cheers
Fritz

**Comment by MehmetGoktay** [comment]
Thank you Fritz!

**Comment by fritzsedlazeck** [comment]
I pushed the code. Please have another try and let me know. 
Thanks
Fritz

**Comment by MehmetGoktay** [comment]
Thank you very much Fritz!
Now it works!

Best,
Mehmet

**Comment by fritzsedlazeck** [comment]
Thanks for letting me know.
Cheers
Fritz

**Comment by hdore** [comment]
Hi Fritz,

I know this has been closed 2 years ago, but I am having the exact same problem with version 1.0.12.
I obtain almost none 0/0 genotypes. I get 1/1 genotypes when there is an SV, and otherwise mostly ./. genotypes with 0 reads supporting the reference allele. In the few examples I have checked on IGV, I see a lot of reads supporting the reference though. 

Maybe it's related (or maybe not), but the number of reads supporting either the reference allele or the variant are very low compared to the coverage and to what I can see in IGV. I thought that might be because of quality thresholds, but using very low quality thresholds in sniffles (--minmapping_qual 1 -r 100) did not really help (I only see a negligible increase in read support compared to what I see in IGV). This is the case both at the de novo and force calling steps of population SV calling.

My data are Nanopore reads aligned to a reference bacterial genome with ngmlr.

Thank you for your help,

Best,

Hugo
